internal/supervisor/installation.go

```diff
@@ -685,3 +689,36 @@ func (s *InstallationSupervisor) finalDeletionCleanup(installation *model.Instal
 
 	return model.InstallationStateDeleted
 }
+
+func (s *InstallationSupervisor) finalCreationTasks(installation *model.Installation, logger log.FieldLogger) string {
+	clusterInstallations, err := s.store.GetClusterInstallations(&model.ClusterInstallationFilter{
+		InstallationID: installation.ID,
+		PerPage:        model.AllPerPage,
+	})
+	if err != nil {
+		logger.WithError(err).Warn("Failed to find cluster installations")
+		return model.InstallationStateCreationFinalTasks
+	}
+
+	for _, clusterInstallation := range clusterInstallations {
+		cluster, err := s.store.GetCluster(clusterInstallation.ClusterID)
+		if err != nil {
+			logger.WithError(err).Warnf("Failed to query cluster %s", clusterInstallation.ClusterID)
+			return model.InstallationStateCreationFinalTasks
+		}
+
+		certName := fmt.Sprintf("%s-tls-cert", strings.Replace(installation.DNS, ".", "-", -1))
+		status, err := s.provisioner.WaitForCertApproved(cluster, installation.ID, certName)
+		if err != nil {
+			logger.WithError(err).Error("Couldn't get the certificate status")
+			return model.InstallationStateCreationFinalTasks
+		}
+		if status == "False" {
```

Comment:
@gabrieljackson The cert status returns a True, False or Unknown status (you can see this [here](https://github.com/jetstack/cert-manager/blob/15d1735688a907c3fde90e5801e86d0e1abbaefe/pkg/apis/certmanager/v1alpha3/types_certificate.go#L206). Do you think we should check for True instead of the False status?

Reply:
Thanks for the info!

Yeah, I think checking for `True` would be better here as right now we assume that if it isn't `False` then we are good to go. I could see there possibly being new status in the future as well so it would probably be better to check specifically for `True` to know things were successful.

What do you think?

Reply:
Makes sense, moved to check for True ;)